service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java

```diff
@@ -222,6 +222,21 @@ private void processSamples(List<Sample> samples, String projection) {
                     sample.setProfiledForFusions(false);
                 }
             });
+        } else if (projection.equals("SEQUENCED")) {
+            Map<String, Set<String>> sequencedSampleIdsMap = new HashMap<>();
+            List<String> distinctStudyIds = samples.stream().map(Sample::getCancerStudyIdentifier).distinct()
+                .collect(Collectors.toList());
+            List<MolecularProfile> molecularProfiles = molecularProfileRepository.getMolecularProfilesInStudies(distinctStudyIds, projection);
+
+            for (String studyId : distinctStudyIds) {
+                sequencedSampleIdsMap.put(studyId,
+                    new HashSet<String>(sampleListRepository.getAllSampleIdsInSampleList(studyId + SEQUENCED)));
+            }
+
+            samples.forEach(sample -> {
+                sample.setSequenced(sequencedSampleIdsMap.get(sample.getCancerStudyIdentifier())
+                    .contains(sample.getStableId()));
```

Comment:
Because sample stable ids are not globally unique, wouldn't we run into a problem in a case like this:
cancer_study `studyX` has molecular_profile `X_mutations` has patient `p24` has sample `s6`
cancer_study `studyX` has molecular_profile `X_mutations` has patient `p24` has sample `s8`
cancer_study `studyY` has molecular_profile `Y_mutations` has patient `p57` has sample `s3`
cancer_study `studyY` has molecular_profile `Y_mutations` has patient `p57` has sample `s8`
If the query list includes [
{"molecular profile" : "X_mutations", "sample_stable_id" : "s8"} , 
{"molecular profile" : "Y_mutations", "sample_stable_id" : "s3"}
]
Is it possible we are going to be returning/counting 3 samples ... X_mutations s8, Y_mutations s3, Y_mutations s8 ?

Reply:
c